File: 56. Merge Intervals/merge.java
class merge {
    public int[][] solution(int[][] intervals) {
        Arrays.sort(intervals, (v1, v2) -> v1[0] - v2[0]);
        int[][] res = new int[intervals.length][2];
        int idx = -1;
        for (int[] i: intervals) {
            if (idx == -1 || i[0] > res[idx][1]) {
                res[++idx] = i;
            } else {
                res[idx][1] = Math.max(res[idx][1], i[1]);
            }
        }
        return Arrays.copyOf(res, idx + 1);
    }
}

File: 56. Merge Intervals/merge.py
class merge:
    def solution(self, intervals):
        intervals.sort()
        i = 0
        while i < len(intervals) - 1:
            if intervals[i][1] >= intervals[i + 1][0]:
                intervals[i][1] = max(intervals[i][1], intervals[i + 1][1])
                intervals.pop(i + 1)
            else:
                i = i + 1
        return intervals
        

arr = [[8,10],[1,3],[2,6],[15,18]]
obj = merge()
res = obj.solution(arr)
print(res)